angular-workspace/projects/ni/nimble-angular/pipes/duration.pipe.ts

```diff
@@ -0,0 +1,29 @@
+import { Inject, LOCALE_ID, Pipe, PipeTransform } from '@angular/core';
+import { DurationFormatter } from '@ni/nimble-components/dist/esm/table-column/duration-text/models/duration-formatter';
+
+/**
+ * A pipe that transforms a number of milliseconds (string/number) into a string formatted like "1 day, 2 hr, 3 min, 4.567 sec"
+ */
+@Pipe({
+    name: 'duration'
+})
+export class DurationPipe implements PipeTransform {
```

Comment:
Note: I just followed [the HLD](https://github.com/ni/nimble/blob/main/packages/nimble-components/src/table/specs/table-column-specs/table-column-duration-text-hld.md?plain=1#L81) for the pipe name and entrypoint of `pipes`.